addons/mail/models/mail_thread.py

```diff
@@ -3585,7 +3585,7 @@ def _notify_by_email_get_base_mail_values(self, message, additional_values=None)
             'mail_message_id': message.id,
             'references': references,
         }
-        if mail_subject != message.subject:
+        if mail_subject != (' '.join(message.subject.splitlines()) if message.subject else message.subject):
             base_mail_values['subject'] = mail_subject
```

Comment:
Doesn't this kind of defeat the purpose of inlineing it earlier if we don't force-inline it?

Reply:
Hello, thank you for your feedback! However, I’m not sure I fully understood 😅

Reply:
Above we have `# replace new lines by spaces to conform to email headers requirements`, but here we're saying if the "corrected mail subject" is "just the original subject, but inlined" we don't want to force that value. Which means we won't be inlining the subject, which the comment says we should be doing.

So the proposition is, wherever the error about not being allowed to modify the subject is generated, we avoid it somehow by specifying the subject may change so long as we're only removing new lines. Rather than changing stuff in mail generation.